biggus/__init__.py

```diff
@@ -561,6 +563,175 @@ def _normalise_keys(self, keys):
         return keys
 
 
+class ArrayContainer(Array):
```

Comment:
I see. Makes some sense. My only defence was that I had just implemented another array container (TransposedArray) and was doing exactly the same stubby properties again, so decided to do them one last time, and then refactor as a separate PR. I don't mind doing all of the refactoring in this PR, but it is already surprisingly long for such a simple concept.